src/Swarm/TUI/Controller.hs

```diff
@@ -636,8 +649,8 @@ updateAchievements = do
 --   etc.).
 runGameTick :: EventM Name AppState ()
 runGameTick = do
-  zoomGameState gameTick
-  updateAchievements
+  ticked <- zoomGameState gameTick
+  when ticked updateAchievements
```

Comment:
Why do this only when `ticked` is true?  Shouldn't we update the achievements the instant one is achieved, even if we're in the middle of stepping the focused robot?

Reply:
The same reasoning as with the objective evaluation, it takes time and should not produce different results.

Imagine there is a robot that goes after you that stops you from what you are trying to achieve. It must have its turn, otherwise, you could briefly win in single-step mode, but never in normal mode.